src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -1798,6 +1798,40 @@
       <summary>Creates an ImplicitStackAllocArrayCreationExpressionSyntax node.</summary>
     </FactoryComment>
   </Node>
+  <Node Name="CollectionExpressionSyntax" Base="ExpressionSyntax">
```

Comment:
i also considered CollectionExpressionCreationExpression, but i felt that was too unweildy.  This is similar to InitializerExpression (which represents `{ a, b, c, }`), so i felt this was a suitable name.  But we can def consider and run by the api group.  Other options include.:

CollectionLiteralExpression (but i felt that Literal didn't help, and also made it confusing give that we have LiteralExpressionSyntax).

Reply:
`ArrayCreationExpression` -> `CollectionCreationExpression`?